utilities.cpp

```diff
@@ -66,3 +66,46 @@ bool are_from_same_distribution(const std::vector<double> &lhs, const std::vecto
   else
     return false;
 }
+
+bool are_same_distribution(std::uniform_real_distribution<double>lhs,  std::uniform_real_distribution<double> rhs)
+{
+  std::vector<double> lhs_cues(1000);
+  std::vector<double> rhs_cues(1000);
```

Comment:
the `1000` here is not so good. This hard typed numbers are called magic numbers and they are to be avoided if possible. Moreover in this case they also have to match the number of repeats so I would just declare a variable repeats and initialize the vectors to be the size of the number of repeats. Honestly you do not even have to initialize the vectors, just push_back() in the loop where you generate random numbers from the distributions and just do away with the initialization, this way you will reduce the sources of possible errors.

Reply:
Maybe give the number of repeats as an argument to the fuinction itself, and default it to 1000